nbconvert/preprocessors/tests/test_execute.py

```diff
@@ -69,14 +72,17 @@ def build_preprocessor(opts):
     return preprocessor
 
 
-def run_notebook(filename, opts, resources):
+def run_notebook(filename, opts, resources, preprocess_notebook=None):
```

Comment:
As described below, you can avoid adding this complication that also introduces an additional meaning to what it means to preprocess a notebook.